### ISSUE TITLE
Split genesis state into per-component genesis states.

### ISSUE THREAD

**Issue by hdevalence** [penumbra-zone/penumbra#3085]
**Is your feature request related to a problem? Please describe.**

As part of #2288, we split up the Penumbra application into a coherent collection of "components" (from its previous state as a big-ball-of-mud). This refactoring split the crates into two rough "categories":

- "base" crates that do parts of the application logic, and can depend on each other as needed (e.g., shielded pool, stake, dex)
- "superstructure" crates that pull in all the base crates and assemble them into a unified whole (e.g., transaction, which has all the actions, app, which has all the components, etc)

The current genesis data mixes categories. We have one [monolithic genesis data structure](https://rustdoc.penumbra.zone/main/penumbra_chain/genesis/enum.AppState.html), which must be accessible to every component, but which also has data relevant to every component.  This would create dependency cycles if we added structured data to it, along the lines of the chain parameters issue in #3084

**Describe the solution you'd like**

Split the genesis data up into per-component genesis data, and have a top-level genesis data in the app crate that contains the genesis data for each component.  This is already partially expressed by the `AppState` associated type on the component trait -- it's just that currently, we always set that to the same monolithic type, rather than a per-component structure.

**Additional context**
A final part of #2288.

**Comment by hdevalence** [comment]
As an example, we currently have something like
```rust
#[serde(try_from = "pb::GenesisContent", into = "pb::GenesisContent")]
pub struct Content {
    /// Global configuration for the chain, such as chain ID and epoch duration.
    pub chain_params: ChainParameters,
    /// The initial validator set.
    pub validators: Vec<pb_stake::Validator>,
    /// The initial token allocations.
    pub allocations: Vec<Allocation>,
}
```
but this would be split up into
```rust
// in app
pub struct GenesisState {
    /// Genesis data and configuration for the shielded pool
    pub shielded_pool: penumbra_shielded_pool::GenesisState,
    /// Genesis data and configuration for the staking component
    pub stake: penumbra_stake::GenesisState,
}
// in shielded pool
pub struct GenesisState {
    pub params: Parameters,
    pub allocations: Vec<Allocation>,
}
// in stake
pub struct GenesisState {
    pub params: Parameters,
    pub validators: Vec<Validator>,
}
```

**Comment by zbuc** [comment]
Closed by #3086
